src/tqec/templates/orchestrator.py

```diff
@@ -48,7 +49,7 @@ def get_corner_position(
     )
 
 
-class TemplateOrchestrator(JSONEncodable):
+class TemplateOrchestrator(Template):
     def __init__(self, templates: list[TemplateWithIndices]) -> None:
```

Comment:
Instead of wrapping a `Template` with an extra `TemplateWithIndices`, I prefer reserving the plaquette indices as a property of the template itself which can be changed with something like `template.with_indices(...)`. This makes the `TemplateOrchestrator` behave more like a composer of `Template`.

Reply:
This is a good remark. Again I would need to think deeper into it, but at first glance the current approach does not bother me too much because:
- it is simple,
- it keeps `Template` as a *description* of something, that can then be instantiated at will with concrete indices,
- the `TemplateWithIndices` structure is not visible from the public API.

I would be fine having a `Template.with_indices(...)` returning a `TemplateWithIndices` instance, as a way of writing that part in a more readable manner.

But storing indices within `Template` instances would break the "a `Template` is a description that can be instantiated at will".

Also I realise that something might be unclear with the current code: the indices provided as second argument to `TemplateWithIndices` are not the plaquette indices that are used to instantiate the `Template` instance given as first argument. They are indices that will be used on the list of plaquette indices provided to the `instanciate` method. So `TemplateWithIndices` is not storing any plaquette indices, only how to index in the list of plaquette indices that will eventually be provided to `instanciate`.
With an example,
https://github.com/QCHackers/tqec/blob/4a2a03d1bd22a3746cadf92114492754c9e6048e/src/tqec/templates/constructions/qubit.py#L24-L25
means that:
- if the `instanciate` method is called with `[0, 1, 2, 3, 4, 5, 6]` (which is nearly always the case), the 3rd and 4th indices from that list will be recovered: `3` and `4`,
- if the `instanciate` method is called with `[0, 2, 4, 6, 1, 3, 5]` (which might make sense in some context), the 3rd and 4th indices from that list will be recovered: `6` and `1`.

So having these indices in the `Template` description (here, `TemplateOrchestrator`) does not break point 2. above: these are not the indices the `Template` will be instantiated with, only a way to know which plaquette indices should be used.